static/js/poll_modal.js

```diff
@@ -0,0 +1,70 @@
+import $ from "jquery";
+import {Sortable} from "sortablejs";
+
+import render_poll_modal_option from "../templates/poll_modal_option.hbs";
+
+export function update_option_delete_btn($container, display_flag) {
+    const no_of_option_row = $container.find(".option-row").length;
+
+    // When there is only one option row, we hide its delete button
+    if (no_of_option_row === 1) {
+        // display_flag indicates whether another row will be added
+        // next, and in that case, we should show the delete button
+        if (display_flag === true) {
+            $container.find(".option-row .delete-option").show();
+        } else {
+            $container.find(".option-row .delete-option").hide();
+        }
+    }
+}
+
+function create_option_row(container) {
+    const row = render_poll_modal_option();
+    $(container).append(row);
+}
+
+function add_option_row(e) {
+    // if the option triggering the input event e is not the last,
+    // that is, its next sibling has the class `option-row`, we
+    // do not add a new option row and return from this function
```

Comment:
Maybe adding a line here that says something like, `This handles a case when the next empty input row is already added and user is updating the above row(s).`  might be more clear.